components/prism-core.js

```diff
@@ -28,7 +28,7 @@ var _ = _self.Prism = {
 			} else if (_.util.type(tokens) === 'Array') {
 				return tokens.map(_.util.encode);
 			} else {
-				return tokens.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/\u00a0/g, ' ');
```

Comment:
Why did you remove this?

Reply:
`<pre>` is supposed to retain newline characters, so it doesn't make sense to remove them (in the original code, see merge request).

Reply:
This line caused text tokens to be altered from `\nTest\n` to ` Test `.

So what was ...

```javascript
var i;

Test

var j;
```

... in the original `<code>` element became ...

```javascript
var i; Test var j;
```

Reply:
See my sample HTML file in issue #1764.

Check the result when you alter the code content to:

```javascript
var i = 1;

Test

function()
{}
```

Reply:
This line only escapes text to be valid HTML, so that e.g. the code `<p>` is not interpreted as a tag but as text.

I'm asking why you removed the removal of No-break spaces. I don't see how this relates to new lines.
(TBH, I don't know why we remove NBSPs in the first place)

Reply:
D'oh! You are right. I misread the `\u00a0` as `\u000a`.

The Firefox debugger I'm using is not very stable. While debugging it turned out that the debugger stripped the new-lines from my test when I stepped over that function. I then examined that function and misread the literal value.

Yet, it's like you wrote: There is no need to escape the non-breaking space. I can remove the change, though, if you prefer.

Reply:
Yes, please remove this change.
It's better to keep things simple.